quic/s2n-quic-platform/src/message/cmsg.rs

```diff
@@ -4,173 +4,248 @@
 #![allow(clippy::unnecessary_cast)] // some platforms encode lengths as `u32` so we cast everything to be safe
 
 use core::mem::{align_of, size_of};
+use libc::cmsghdr;
 use s2n_quic_core::inet::{AncillaryData, ExplicitCongestionNotification};
 
+const fn size_of_cmsg<T: Copy + Sized>() -> usize {
+    unsafe { libc::CMSG_SPACE(size_of::<T>() as _) as _ }
+}
+
+const fn const_max(a: usize, b: usize) -> usize {
+    if a > b {
+        a
+    } else {
+        b
+    }
+}
+
 /// The maximum number of bytes allocated for cmsg data
 ///
 /// This should be enough for UDP_SEGMENT + IP_TOS + IP_PKTINFO. It may need to be increased
 /// to allow for future control messages.
-pub const MAX_LEN: usize = 128;
+pub const MAX_LEN: usize = {
+    let tos_size = size_of_cmsg::<IpTos>();
+
+    #[cfg(s2n_quic_platform_gso)]
+    let gso_size = size_of_cmsg::<UdpGso>();
+    #[cfg(not(s2n_quic_platform_gso))]
+    let gso_size = 0;
+
+    #[cfg(s2n_quic_platform_gro)]
+    let gro_size = size_of_cmsg::<UdpGro>();
+    #[cfg(not(s2n_quic_platform_gro))]
+    let gro_size = 0;
+
+    let segment_offload_size = const_max(gso_size, gro_size);
+
+    #[cfg(s2n_quic_platform_pktinfo)]
+    let pktinfo_size = const_max(
+        size_of_cmsg::<libc::in_pktinfo>(),
+        size_of_cmsg::<libc::in6_pktinfo>(),
+    );
+    #[cfg(not(s2n_quic_platform_pktinfo))]
+    let pktinfo = 0;
+
+    // This is currently needed due to how we detect if CMSG data has been written or not.
+    //
+    // TODO remove this once we split the `reset` traits into TX and RX types
```

Comment:
Are you tracking this somewhere?

Reply:
It's in my later PRs so it's already done; just is waiting on deleting the socket v1